core/primitives/src/lib.rs

```diff
@@ -11,15 +11,18 @@ pub mod account;
 pub mod block;
 pub mod block_header;
 pub mod challenge;
+pub mod config;
 pub mod contract;
 pub mod epoch_manager;
 pub mod errors;
 pub mod hash;
 pub mod logging;
 pub mod merkle;
 pub mod network;
+pub mod profile;
 pub mod receipt;
 pub mod rpc;
+pub mod runtime;
```

Comment:
Our core primitives are already enormous (it has 35 flat modules in the root). What is the driving factor for collapsing even more crates into it?

Reply:
Lifting deps that are across multiple crates.  E.i. there should only be one set of primitive types.  I solved this by adding a new crate `near-primitives-core` which can be depended on by both `near-primitives` and vm crates.

Reply:
I love that we remove code duplication! @willemneal thanks for championing this!

@evgenykuzyakov do we have a test in nearcore or near-sdk-rs that confirms that it is compilable with the latest `near-vm-logic` and `near-vm-errors`?

We need to confirm that `near-sdk-rs` is not going to break after this refactoring.